codec/src/main/java/io/netty/handler/codec/xml/XmlFrameDecoder.java

```diff
@@ -58,174 +58,295 @@
+                if (c == '>' && in.getByte(idx - 1) == '?') {
+                    stat = lastStat;
+                }
+            } else if (stat == XmlState.COMMENT) {
+                if (c == '-') {
+                    if ((idx + 2) >= wrtIdx) {
+                        return;
+                    }
+                    // closed the xml comment
+                    if (in.getByte(idx + 1) == '-'
+                            && in.getByte(idx + 2) == '>') {
+                        idx += 2;
+                        stat = lastStat;
+                    }
+                }
+            } else if (stat == XmlState.HEAD) {
+                // found the attribute <XmlNodeid="nodeid"></XmlNode>
+                if (c == '=') {
+                    if ((idx + 1) >= wrtIdx) {
+                        return;
+                    }
+                    if (in.getByte(idx + 1) == '"') {
+                        idx++;
+                        lastStat = stat;
+                        stat = XmlState.ATTR;
+                    }
+                } else if (c == '>') {
+                    // found the closed tag node head
+                    // found the closed tag in head like <Xml />
+                    if (in.getByte(idx - 1) == '/') {
+                        openCount--;
+                    }
+                    stat = XmlState.BODY;
+                }
+            } else if (stat == XmlState.ATTR) {
+                // found closed attribute
+                if (c == '"' && in.getByte(idx - 1) != '\\') {
+                    stat = lastStat;
+                }
+            } else if (stat == XmlState.BODY) {
+                if (c == '<') {
+                    if ((idx + 1) >= wrtIdx) {
+                        return;
+                    }
+                    byte c_next = in.getByte(idx + 1);
+                    if (c_next == '/') {
+                        closeing = true;
+                        stat = XmlState.END;
+                        idx++;
+                    } else if (c_next == '!') {
+                        if ((idx + 3) >= wrtIdx) {
+                            return;
+                        }
+                        // <!-- comment --> start found
+                        if (in.getByte(idx + 2) == '-'
+                                && in.getByte(idx + 3) == '-') {
+                            lastStat = stat;
+                            stat = XmlState.COMMENT;
+                            idx += 3;
+                            continue;
+                        }
+                        if ((idx + 8) >= wrtIdx) {
+                            return;
                         }
-                    } else if (peekAheadByte == '?') {
-                        // <?xml ?> start found
-                        openBracketsCount++;
+                        // <![CDATA[ start found
+                        if (in.getByte(idx + 2) == '['
+                                && in.getByte(idx + 3) == 'C'
```

Comment:
Please move `&&` to the end of the previous lines. Please fix the same formatting issues in this changeset.